presto-native-execution/presto_cpp/main/types/PrestoToVeloxConnector.cpp

```diff
@@ -1075,39 +1107,34 @@ HivePrestoToVeloxConnector::toVeloxSplit(
   infoColumns.insert(
       {"$file_modified_time",
        std::to_string(hiveSplit->fileSplit.fileModifiedTime)});
-  return std::make_unique<velox::connector::hive::HiveConnectorSplit>(
-      catalogId,
-      hiveSplit->fileSplit.path,
-      toVeloxFileFormat(hiveSplit->storage.storageFormat),
-      hiveSplit->fileSplit.start,
-      hiveSplit->fileSplit.length,
-      partitionKeys,
-      hiveSplit->tableBucketNumber
-          ? std::optional<int>(*hiveSplit->tableBucketNumber)
-          : std::nullopt,
-      customSplitInfo,
-      extraFileInfo,
-      serdeParameters,
-      hiveSplit->splitWeight,
-      infoColumns);
+  auto veloxSplit =
+      std::make_unique<velox::connector::hive::HiveConnectorSplit>(
+          catalogId,
+          hiveSplit->fileSplit.path,
+          toVeloxFileFormat(hiveSplit->storage.storageFormat),
+          hiveSplit->fileSplit.start,
+          hiveSplit->fileSplit.length,
+          partitionKeys,
+          hiveSplit->tableBucketNumber
+              ? std::optional<int>(*hiveSplit->tableBucketNumber)
+              : std::nullopt,
+          customSplitInfo,
+          extraFileInfo,
+          serdeParameters,
+          hiveSplit->splitWeight,
+          infoColumns);
+  if (hiveSplit->bucketConversion) {
+    veloxSplit->bucketConversion =
+        toVeloxBucketConversion(*hiveSplit->bucketConversion);
+  }
+  return veloxSplit;
 }
 
 std::unique_ptr<velox::connector::ColumnHandle>
 HivePrestoToVeloxConnector::toVeloxColumnHandle(
     const protocol::ColumnHandle* column,
     const TypeParser& typeParser) const {
-  auto hiveColumn = dynamic_cast<const protocol::HiveColumnHandle*>(column);
-  VELOX_CHECK_NOT_NULL(
-      hiveColumn, "Unexpected column handle type {}", column->_type);
-  velox::type::fbhive::HiveTypeParser hiveTypeParser;
-  // TODO(spershin): Should we pass something different than 'typeSignature'
-  // to 'hiveType' argument of the 'HiveColumnHandle' constructor?
-  return std::make_unique<velox::connector::hive::HiveColumnHandle>(
-      hiveColumn->name,
-      toHiveColumnType(hiveColumn->columnType),
-      stringToType(hiveColumn->typeSignature, typeParser),
-      hiveTypeParser.parse(hiveColumn->hiveType),
-      toRequiredSubfields(hiveColumn->requiredSubfields));
+  return toVeloxHiveColumnHandle(column, typeParser);
```

Comment:
There is no logic change to this function? Thanks!

Reply:
No any logical change to existing code